go/vt/worker/topo_utils.go

```diff
@@ -128,17 +129,18 @@ func FindWorkerTablet(ctx context.Context, wr *wrangler.Wrangler, cleaner *wrang
 	// ChangeSlaveType back, so we need to record this tag change after the change
 	// slave type change in the cleaner.
 	defer wrangler.RecordTabletTagAction(cleaner, tabletAlias, "worker", "")
```

Comment:
I don't like that these Tablet record changes are not atomic. We update the tablet record 2 times upon start, and 3 times upon cleaning. The extra tag makes it worse, but the problem was already there. I guess we can fix that when we switch to having an RPC to make a tablet drained.